src/resdk/resources/metadata.py

```diff
@@ -117,6 +121,9 @@ def set_index(self, df):
         # Remove the samples that do not have mapping
         df = df.dropna(subset=["Sample ID"])
 
+        # We don't need the sample identifier column anymore

```

Comment:
I would skip this - the removal part. I can understand that you do not need this column anymore in your current use cases, and the similarity with `DataFrame.set_index()` method behaviour. However, IMO i't a bit of an overkill to remove e.g. "Sample name" column... For some use cases it might come handy. I would rather keep it and let users remove them themselves if necessary that having to to some jujitsu to keep it in there.